Sources/NIOCore/ByteBuffer-hex.swift

```diff
@@ -340,4 +354,55 @@ extension ByteBuffer {
             }
         }
     }
+
+    public struct HexDecodingError: Error {
+        let kind: HexDecodingErrorKind
+
+        public enum HexDecodingErrorKind {
```

Comment:
This `enum` needs to be private, or we can't safely use this type.

Reply:
Are you suggesting internal? Private would mean this would be impossible (to my understanding) to initialise (atleast the way I have done so).

Reply:
Private should be fine, you'd use `public static let`s on the parent to access the fields:

```swift
public struct HexDecodingError: Error {
    private let kind: Kind

    enum Kind {
        case invalidHexLength
        case invalidCharacter
    }
    
    public static let invalidHexLength = HexDecodingError(kind: .invalidHexLength)
    public static let invalidCharacter = HexDecodingError(kind: .invalidCharacter)
}
```

Reply:
thanks will find time to implement.